### ISSUE TITLE
Review and add required hooks

### ISSUE THREAD

**Issue by darrenmeehan** [Darrenmeehan/BuddyPress-Like#44]
As per #43 I'd like to work on improving the hooks throughout the plugin. On first thought:
1. Hooks before and after liking/unliking an item.
2. Before and after displaying who likes an item.
3. Should we have hooks in all functions, as a requirement?

I'd done some work on hooks, trying to mimic their usage from BuddyPress Core and WordPress. I've to look into any formalised guidelines or requirements from either of these. As a side note, eventually I'd like the plugin to follow any guidelines from either of these, that make sense.

**Comment by fried-eggz** [comment]
I think you should have a hook on every function that performs an action like you describe in in 1 and 2.

BuddyPress uses a lot of before and after hooks, maybe thats a nice starting point.

Also, adding filters in places where data is outputted is nice since it offers a good way of customizing outputs.

Good luck with this. If you need help I'm willing to contribute. I love your work on this plugin,

**Comment by darrenmeehan** [comment]
Yes filters is something else I'd like to work on, once the functionality for the component, posts and notifications is near finished this will be a priority.
